_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,249 @@
+---
+layout: default
+title: Model serving framework 
+has_children: false
+nav_order: 110
+---
+
+# Model serving framework
+
+ML commons supports the ability to serve custom models and use those models to make inferences. For those who wish to use train their models using a deep-learning framework, such as Pytorch, TensorFlow, or ONNX, they can upload such models to OpenSearch in order to take advantage of OpenSearch resources, like GPU acceleration.
+
+This page outlines the steps required to upload a custom model with the ML Commons plugin.
+
+## Prerequisites 
+
+To upload a custom model to OpenSearch, you'll need to prepare outside of your OpenSearch cluster. You can use a current model or train a new model depending on your needs.
+
+### Model format
+
+Although most ML models are written in Python, you'll need to export the model into a portable format. As of 2.4, OpenSearch only supports [torchscript](https://pytorch.org/docs/stable/jit.html).
+
+### Model size
+
+Most deep-learning models are over 100 MBs, making it difficult to fit the model into a single document. OpenSearch splits in portable model file into smaller chunks to store in an OpenSearch index. When allocating ML or Data nodes to your OpenSearch cluster, please be aware of the size of your model to prevent any downtime when making inferences. Furthermore, asynchronous models load faster than synchronous.
```

Comment:
The context is ambiguous. What is an asynchronous versus synchronous model?  Are you talking about the model or the import command? I have only heard of async and sync interfaces for model servers (not models).

Reply:
My understanding rereading the materials provided is that synchronous models need to be immediately loaded immeaditely after upload, otherwise, the upload API only supports asynchronous models.

Reply:
I think we can remove that async/sync part here.